flare_portal/api/views.py

```diff
@@ -14,9 +15,19 @@ def post(self, request: Request, format: str = None) -> Response:
         form = ConfigurationForm(request.data)
 
         if form.is_valid():
-            experiment: Experiment = form.cleaned_data["participant"].experiment
+            participant = form.cleaned_data["participant"]
+            experiment: Experiment = participant.experiment
+
+            # Mark that the participant has started the experiment
+            participant_started_at = participant.started_at
+            if participant.started_at == None:
+                participant.started_at = timezone.now()
+                participant.save()
```

Comment:
Suggestion: The view code shouldn't have business logic.

I would refactor this and put this code in `form.save()`.

Reply:
Refactored out.